### PR TITLE
Add mkdocs.yml and fix CircleCI

### PR DESCRIPTION
**Pull request by adrw**
* Docs build requires it, forgot to copy in the docs site addition

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+site_description: "Backfila: Service that manages backfill state, calling into other services to do batched work"
+site_author: Square, Inc.
+remote_branch: gh-pages
+
+copyright: 'Copyright &copy; 2019 Square, Inc.'
+
+theme:
+  name: 'material'
+  favicon: img/favicon.ico
+  logo: img/cashapp.png
```
Comment (mpawliszyn) [inline mkdocs.yml]: backfila.png?

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  - pymdownx.magiclink
+  - pymdownx.smartsymbols
+  - pymdownx.superfences
+  - tables
+
+nav:
+  - 'Overview': index.md
+  - '0.x API':
+      - 'client': 0.x/client/index.md
+      - 'client-service': 0.x/client-service/index.md
```
Comment (mpawliszyn) [inline mkdocs.yml]: Are these all right?

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+site_name: Backfila
+repo_name: Backfila
+repo_url: https://github.com/cashapp/backfila
+site_description: "Backfila: Service that manages backfill state, calling into other services to do batched work"
+site_author: Square, Inc.
```
Comment (mpawliszyn) [inline mkdocs.yml]: Cashapp?

Comment (adrw) [reply mkdocs.yml]: We still use Square corporate name on all open source legal for now

File: mkdocs.yml (hunk 2 above)
Comment (adrw) [reply mkdocs.yml]: Yes, they match the Gradle subproject names (which could be changed later to make it clearer ie. backfila-client, backfila-client-service, backfila-service)

File: mkdocs.yml (hunk 1 above)
Comment (adrw) [reply mkdocs.yml]: Nice, will add 👍

Comment (adrw) [reply mkdocs.yml]: I'll do this in a follow up, better to land this and unbreak CircleCI

Comment (mpawliszyn) [reply mkdocs.yml]: 🤔

File: mkdocs.yml (hunk 2 above)
Comment (mpawliszyn) [reply mkdocs.yml]: ACK

**Comment by shellderp** [comment]
lg

**Pull request merged**